### PR TITLE
Add snapshot test functionalities

### PR DESCRIPTION
**Pull request by phihag**
Add a helper function for snapshot tests.

### CODE REVIEW

File: bx_py_utils/test_utils/snapshot.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    assert re.match(r'^[-_.a-zA-Z0-9]+$', snapshot_name), f'Invalid snapshot name {snapshot_name}'
+    assert isinstance(got, (dict, list))
+
+    snapshot_file = pathlib.Path(root_dir) / f'{snapshot_name}.json'
+    try:
+        with snapshot_file.open('r') as snapshot_handle:
+            expected = json.load(snapshot_handle)
+    except (ValueError, IOError, FileNotFoundError, OSError):
+        with snapshot_file.open('w') as snapshot_handle:
+            json.dump(got, snapshot_handle)
```
Comment (jedie) [inline bx_py_utils/test_utils/snapshot.py:17]: What's about to add: `ensure_ascii=False, sort_keys=True, indent=4` ?

Same below or better merge into a function?

File: bx_py_utils_tests/tests/test_test_utils_snapshot.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+import pytest
+
+from bx_py_utils.test_utils.snapshot import assert_snapshot
+
+
+def test_assert_snapshot():
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        with pytest.raises(FileNotFoundError) as excinfo:
+            assert_snapshot(tmp_dir, 'snap', [{'foo': 42, 'bar': 5}])
```
Comment (jedie) [inline bx_py_utils_tests/tests/test_test_utils_snapshot.py:12]: add a non ascii example, too.

File: bx_py_utils/test_utils/snapshot.py (hunk 1 above)
Comment (phihag) [reply bx_py_utils/test_utils/snapshot.py:17]: Very good points, totally forgot! Added all.

File: bx_py_utils_tests/tests/test_test_utils_snapshot.py (hunk 2 above)
Comment (phihag) [reply bx_py_utils_tests/tests/test_test_utils_snapshot.py]: Updated the boring `bar` to an exciting `bär`

**Review by jedie — APPROVED** [review]

**Pull request merged**